pkg/backend/display/events.go

```diff
@@ -182,19 +184,18 @@ func convertStepEventMetadata(md engine.StepEventMetadata) apitype.StepEventMeta
 	}
 }
 
-func convertStepEventStateMetadata(md *engine.StepEventStateMetadata) *apitype.StepEventStateMetadata {
+func convertStepEventStateMetadata(
+	md *engine.StepEventStateMetadata, encrypter config.Encrypter) *apitype.StepEventStateMetadata {
+
 	if md == nil {
 		return nil
 	}
 
-	inputs := make(map[string]interface{})
-	for k, v := range md.Inputs {
-		inputs[string(k)] = v
-	}
-	outputs := make(map[string]interface{})
-	for k, v := range md.Outputs {
-		outputs[string(k)] = v
-	}
+	inputs, err := stack.SerializeProperties(md.Inputs, encrypter)
+	contract.IgnoreError(err)
+
+	outputs, err := stack.SerializeProperties(md.Outputs, encrypter)
+	contract.IgnoreError(err)
```

Comment:
Who were existing consumers of this data?  What expectations did they have?  Seems clear this data was "wrong" beforehand, but would be good to understand consumer sites to understand what if anything we need to do in terms of compatibility here.

Also - can we add tests that lock down the shape of this event metadata?

Reply:
The service is the only existing consumer of this data. I asked @chrsmith above about the service's usage of these maps; he's looking into it.